### DISCUSSION TITLE
Add HDR support

### DISCUSSION THREAD

**Discussion by vbataille76** [LizardByte/community#82] [Sunshine] [answered]
### Describe the Bug

As soon as HDR is enabled in Windows 11 settings, sunshine server won't start.
How to reproduce:
1) Enable HDR in W11 Settings (System->Display).
2) Start Sunshine. it stops immediately.

Note: It starts and works perfectly with HDR disabled.  If HDR is enabled after starting the server, server stops as soon as streaming is started.

### Expected Behavior

Sunshine should work with and without HDR toggle enabled.

### Host Operating System

Windows

### Operating System Version

11 professional (22622.598), 22H2

### Architecture

64 bit

### Sunshine Version

0.14.1

### GPU Type

NVidia

### GPU Model

2080 TI

### GPU Driver/Mesa Version

516.94

### Relevant log output

```Shell
[fps] -- [[30,60,90,120]]
[vt_coder] -- [auto]
[amd_rc] -- [auto]
[vt_realtime] -- [enabled]
[key_rightalt_to_key_win] -- [enabled]
[resolutions] -- [[
    1280x720,
    1920x1080,
    3860x2160,
    2560x1440
]]
[origin_pin_allowed] -- [pc]
[gamepad] -- [x360]
[amd_coder] -- [auto]
[adapter_name] -- [NVIDIA GeForce RTX 2080 Ti]
[nv_preset] -- [default]
[upnp] -- [disabled]
[nv_coder] -- [auto]
[dwmflush] -- [enabled]
[output_name] -- [\\.\DISPLAY1]
[min_log_level] -- [0]
[vt_software] -- [auto]
[amd_quality] -- [default]
[origin_web_ui_allowed] -- [lan]
[nv_rc] -- [auto]
[hevc_mode] -- [0]
[2022:09:22:20:09:53]: Info: Compiling shaders...
[2022:09:22:20:09:53]: Info: Compiled shaders
[2022:09:22:20:09:53]: Info: //////////////////////////////////////////////////////////////////
[2022:09:22:20:09:53]: Info: //                                                              //
[2022:09:22:20:09:53]: Info: //   Testing for available encoders, this may generate errors.  //
[2022:09:22:20:09:53]: Info: //   You can safely ignore those errors.                        //
[2022:09:22:20:09:53]: Info: //                                                              //
[2022:09:22:20:09:53]: Info: //////////////////////////////////////////////////////////////////
[2022:09:22:20:09:53]: Info: Trying encoder [nvenc]
[2022:09:22:20:09:53]: Info:
Device Description : NVIDIA GeForce RTX 2080 Ti
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00001E07
Device Video Mem   : 11049 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 32698 MiB
Feature Level      : 0x0000B100
Capture size       : 3840x2160
Offset             : 3840x0
Virtual Desktop    : 7680x2160
[2022:09:22:20:09:53]: Debug: Source format [DXGI_FORMAT_R16G16B16A16_FLOAT]
```

**Comment by ReenigneArcher** [comment]
I don't know that this should be an "issue" (a.k.a bug report), but probably instead should be a feature request, so I will transfer it there.

⚠️ There is no reason to add `+1` or `I have this issue too`. It doesn't work for anyone. ⚠️

**Reply by vbataille76** [reply]
I understand that this can be considered as a feature. However, I would expect the server to run (at least streaming SDR content), instead of stopping without further explanation. Maybe it's not possible though.

**Reply by blackgold9** [reply]
Thanks for your work here.  I "worry" that marking this a feature request for future development vs a bug is a miscategorization.

I think the feature request is:
"Stream HDR"

The bug here is;
"It crashes when HDR is on". 

The bugfix could be as simple as an error message stating HDR is not supported rather than a crash.

This is of course just my 2 cents.

**Reply by ReenigneArcher** [reply]
> I understand that this can be considered as a feature. However, I would expect the server to run (at least streaming SDR content), instead of stopping without further explanation. Maybe it's not possible though.

This is the current behavior in v0.17.0. HDR is still in process though.

**Comment by Gorehound88** [comment]
Hello, I would like to add the same issue. Please look into this issue, as the server crashes immediately when HDR is enabled in Windows 11.

**Comment by iopean** [comment]
I have this very same issue too. The Main 10 codec works fine on Gamestream but if HDR is enabled in Windows then Sunshine fails to start.

**Comment by nikc0069** [comment]
I also have this issue. This becomes important with the sunsetting of Gamestream in GFE.

**Comment by cgutman** [comment, marked as answer]
Here is an HDR-capable build for testing from #825: https://github.com/LizardByte/Sunshine/suites/10542808200/artifacts/524262833

The only known issue is that the brightness of the mouse cursor may not line up with the actual brightness on the host PC when streaming HDR.

Let me know how it works for you.

**Reply by blackgold9** [reply]
Gave it a spin and it went mostly well. Moonlight rendered it as HVEC Main 10 HDR, but while it rendered well on the host screen, on my MBP 14 it showed in the washed out look that you get when an hdr signal isn't tone mapped right.

**Reply by cgutman** [reply]
That may be because your host's HDR calibration is set to its own display capabilities rather than those of your client's display. You may try streaming the Windows HDR Calibration app to your client and creating a calibration profile for your client and/or adjusting HDR settings in game.

**Reply by laernumoretz** [reply]
> Here is an HDR-capable build for testing from #825: https://github.com/LizardByte/Sunshine/suites/10542808200/artifacts/524262833
> 
> The only known issue is that the brightness of the mouse cursor may not line up with the actual brightness on the host PC when streaming HDR.
> 
> Let me know how it works for you.

Freaking awesome man.  My LG from the shield picked up the HDR

now hopefully it doesnt make my 2080 ti crash like gamestream + new NV drivers was lol.

Only thing I notice is that both hdr and sdr looks exactly the same on my OLED tv. Colors doent seem to have that boosted hdr effect for some reason. Was this just to test the signal itself?

**Reply by Ariane-B** [reply]
I gave it a try, hoping to put my TV's HDR capabilities to use, but it didn't push an HDR signal. Kinda obvious, now that I think about it: my TV is the only HDR-capable "monitor" I own and it's not plugged into the PC.

But it's making me wonder, is it considered for Sunshine to at some point be able to push HDR, regardless of what's plugged into the host PC? I don't know if it would require you to create a "virtual monitor" or something, but it would certainly be nice. I can't see a lot of people splurging on an HDR monitor if the thing they plan to play the most on is their TV, after all.

**Reply by sjvanterpool** [reply]
> 

I have an EDID emulator that pretends to be a monitor with HDR support plugged in alongside my actual monitor. It is disabled unless the main monitor is turned off. That's how I trick the PC into generating an HDR signal.

**Reply by Ariane-B** [reply]
> > 
> 
> I have an EDID emulator that pretends to be a monitor with HDR support plugged in alongside my actual monitor. It is disabled unless the main monitor is turned off. That's how I trick the PC into generating an HDR signal.

Hmm. I only have 1 HDMI port on my graphics card. Would that be a problem?

**Reply by sjvanterpool** [reply]
You can get a display port to HDMI adapter

**Comment by nikc0069** [comment]
Triggers hdr on my shield plugged into my cx perfectly! This is great!

[a comment by an automated account is not shown]
